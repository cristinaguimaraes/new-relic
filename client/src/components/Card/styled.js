import styled from 'styled-components';

export const Host = styled.div`
font-size: 16px;
font-family: 'Helvetica Bold';
color: #000000;
line-height: 19px;
margin-bottom: 23px;
`;

export const App = styled.div`
display: flex;
  :hover {
      cursor: pointer;
    };
 font-family: 'Helvetica Bold';
`;

export const Wrapper = styled.div`
display:flex;
width:${props => (props.layout === 'grid' ? '315px' : '780px')};
min-width: ${props => (props.layout === 'grid' ? '315px' : '780px')};
max-width: ${props => (props.layout === 'grid' ? '315px' : '780px')};
padding: 30px;
flex-direction: column;
background-color: white;
margin-top: 30px;
margin-left: ${props => ((props.layout === 'grid' && props.index % 2) ? '30px' : '0px')};
`;

export const Apdex = styled.div`
color: #4A4A4A;
opacity: 62%;
font-size: 13px;
font-family: 'Helvetica Bold';
line-height: 19px;
min-width: 22px;
`;

export const Name = styled.div`
color: #4A4A4A;
font-size: 16px;
font-family: 'Helvetica';
line-height: 19px;
margin-left: 10px
`;
