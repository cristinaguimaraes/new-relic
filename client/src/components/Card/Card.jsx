import React from 'react';
import { oneOf, string, number, shape, arrayOf } from 'prop-types';
import {Host, App, Wrapper, Name, Apdex} from './styled';

 const Card = ({host, apps, layout}) => {


    return (
        <Wrapper layout={layout}>
            <Host>  {host} </Host>
                {apps.map(app => (
                    <App>
                    <Apdex> {app.apdex} </Apdex><Name> {app.name}</Name>
                    </App>
                    ))}
        </Wrapper>
    )
};

Card.propTypes = {
    host: string.isRequired,
    layout: oneOf(['list', 'grid']).isRequired,
    apps: arrayOf(shape({apdex: number.isRequired, name: string.isRequired})).isRequired
};

export { Card};