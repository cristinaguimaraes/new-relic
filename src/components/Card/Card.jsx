import React from "react";
import { oneOf, string, number, shape, arrayOf, func } from "prop-types";
import { Host, App, Wrapper, Name, Apdex } from "./styled";

const Card = ({ host, apps, layout, index, onShowToastMessage }) => (
  <Wrapper layout={layout} index={index}>
    <Host> {host} </Host>
    {apps.map((app, i) => (
      <App
        key={i}
        onClick={() =>
          onShowToastMessage({ messageKey: `Release number ${app.version}`, type:'success' })
        }
      >
        <Apdex> {app.apdex} </Apdex>
        <Name> {app.name}</Name>
      </App>
    ))}
  </Wrapper>
);

Card.defaultProps = {
  layout: "list"
};

Card.propTypes = {
  host: string.isRequired,
  layout: oneOf(["list", "grid"]),
  apps: arrayOf(
    shape({
      apdex: number.isRequired,
      name: string.isRequired,
      version: number.isRequired
    })
  ).isRequired,
  index: number,
  onShowToastMessage: func.isRequired
};

export { Card };
