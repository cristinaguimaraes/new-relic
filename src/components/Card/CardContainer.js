import { Card } from "./Card";
import { onShowToastMessage } from "redux/actions/creators";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

const mapDispatchToProps = {
  onShowToastMessage
};

export const CardContainer = withRouter(
  connect(null, mapDispatchToProps)(Card)
);
