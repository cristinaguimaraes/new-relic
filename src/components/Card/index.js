import { CardContainer } from "./CardContainer";
export { CardContainer as Card };
