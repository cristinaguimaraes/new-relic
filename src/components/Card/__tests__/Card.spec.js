import { Card } from "../Card";
import { shallow } from "enzyme";
import React from "react";

describe("Card component", () => {
  test("should render properly", () => {
    expect(
      shallow(
        <Card
          onShowToastMessage={jest.fn()}
          index={1}
          layout={"list"}
          host={"7e6272f7-098e.dakota.biz"}
          apps={[
            {
              name: "Awesome Wooden Sausages - Schaefer - Hegmann Inc",
              apdex: 100
            }
          ]}
        >
          Test
        </Card>
      )
    ).toMatchSnapshot();
  });
});
