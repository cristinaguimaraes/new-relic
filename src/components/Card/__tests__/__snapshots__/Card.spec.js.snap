// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render properly 1`] = `ShallowWrapper {}`;
